Frontend/src/App.tsx

```diff
@@ -1,163 +1,45 @@
-
-      //get on-chain proposal data
-      const proposals = await daoContract.getAllProposals();
-      setProposals(proposals);
-      console.log("First proposal", proposals[0]);
-    } else {
-      alert(
-        "DAO got talent is in development! Please connect to rinkeby test network to access website."
-      );
-    }
-  };
-
-  //triggers login and creates connection to dao contract
-  const connectWallet = async () => {
-    if (typeof window.ethereum !== "undefined") {
-      await requestAccount(); // login
-      await connectToContracts(); //connect to DAO contract
-    }
-  };
-
-  //rendered component
   return (
-    <Router>
-      <VStack h="full">
-        {/* HEADER */}
-        <Flex
-          w="100%"
-          justify="space-between"
-          h="60px"
-          zIndex={1000}
-          px={8}
-          background="linear-gradient(rgba(255, 255, 255, 0.1), transparent)"
-          backdropFilter="blur(8px)"
-          alignItems="center"
-        >
-          <Flex alignItems="center" gap={4}>
-            <Box w={6}>
-              <Image src={logo} alt="logo" />
-            </Box>
-            <Link to="/">DAOs got talent</Link>
-          </Flex>
-
-          {walletAddress !== "" ? (
-            <Box className="metamask">Logged in as {walletAddress}</Box>
-          ) : (
-            <Box className="metamask" onClick={connectWallet}>
-              Wallet login
-            </Box>
-          )}
-        </Flex>
-
-        {/* MAIN */}
-        <Flex w="full" h="calc(100% - 60px)" px={8} pt={4}>
-          {/* NAVIGATION */}
-          <Flex w={60} h="full" flexDir="column" gap={6}>
-            <Link to="/vote">
-              <Flex alignItems="center" gap={4}>
-                <Box w={6}>
-                  <Image src={vote} alt="logo" />
-                </Box>
-                Vote
-              </Flex>
-            </Link>
-            <Link to="/my-collections">
-              <Flex alignItems="center" gap={4}>
-                <Box w={6}>
-                  <Image src={mycollections} alt="logo" />
-                </Box>
-                My Collections
-              </Flex>
-            </Link>
-          </Flex>
-          <Box mr={10} h="90%">
-            <Divider orientation="vertical" />
-          </Box>
-
-          {/* CONTENT */}
-          <Flex w="full" h="full" justifyContent="center">
-            <Routes>
-              <Route path="/" element={<Home />} />
-              <Route
-                path="/vote"
-                element={<Vote daoContract={daoContract} />}
-              />
-              <Route
-                path="/my-collections"
-                element={<MyCollections daoContract={daoContract} />}
-              />
-            </Routes>
-          </Flex>
-        </Flex>
-      </VStack>
-    </Router>
-  );
+    <BrowserRouter>
+      <StateContext.Provider
+        value={{
+          daoContract,
+          setDaoContract,
+        }}
```

Comment:
Interesting. I don't know that much React. Why are we adding only daoContract and not the other variables such as collections and so on? Why do we need daoContract in the state context?

Thanks for clarifying

Reply:
we don't need to instantiate the contract for every call / transaction instead can use the state var for it. That was the idea behind it. Collections and proposals should be fetched on page change as mentioned above otherwise we'll have old state.

Reply:
when thinking about it, maybe it makes sense to keep the collection data in state, and just fetch for more proposals and add new proposal data also to the collection. However, would do this as next step.

Reply:
ah ok, gotcha. Because proposals and collections and winning collection are mutable. Great